components/brave_wallet/browser/swap_service.cc

```diff
@@ -94,6 +87,80 @@ std::string GetAffiliateAddress(const std::string& chain_id) {
   return "";
 }
 
+brave_wallet::mojom::BraveSwapFeeResponsePtr GetBraveFeeInternal(
+    brave_wallet::mojom::BraveSwapFeeParamsPtr params) {
+  const auto& percentage_fee = GetFeePercentage(params->chain_id);
+  if (!percentage_fee) {
+    return nullptr;
+  }
+
+  if (IsSolanaNetworkSupported(params->chain_id)) {
+    brave_wallet::mojom::BraveSwapFeeResponsePtr response =
+        brave_wallet::mojom::BraveSwapFeeResponse::New();
+
+    bool has_fees =
+        brave_wallet::HasJupiterFeesForTokenMint(params->output_token);
+
+    // Jupiter API v6 will take 20% of the platform fee charged by integrators.
+    // TODO(onyb): update the multipliers below during migration to Jupiter API
+    // v6.
+    auto protocol_fee_pct = percentage_fee.value() * 0;
+    auto brave_fee_pct = percentage_fee.value() * 1;
+    auto discount_on_brave_fee_pct = has_fees ? 0 : 100;
+    auto discount_on_prototcol_fee_pct = has_fees ? 0 : 100;
+
+    response->brave_fee_pct = base::NumberToString(brave_fee_pct);
+
+    response->discount_on_brave_fee_pct =
+        base::NumberToString(discount_on_brave_fee_pct);
+
+    response->protocol_fee_pct = base::NumberToString(
+        (100 - discount_on_prototcol_fee_pct) / 100 * protocol_fee_pct);
+
+    response->effective_fee_pct = base::NumberToString(
+        (100 - discount_on_brave_fee_pct) / 100 * brave_fee_pct);
+
+    // Jupiter swap fee is specified in basis points
+    response->fee_param =
+        base::NumberToString(static_cast<int>(percentage_fee.value() * 100));
+
+    response->discount_code =
+        has_fees ? brave_wallet::mojom::DiscountCode::kNone
+                 : brave_wallet::mojom::DiscountCode::kUnknownJupiterOutputMint;
```

Comment:
`(100 - discount_on_prototcol_fee_pct) / 100 * protocol_fee_pct)` works as expected only when `discount_on_prototcol_fee_pct` is either `0` or `100` - division is done in integers.
Let's force doubles everywhere like `0.0` and `100.0`